src/main/java/org/folio/service/transactions/OpenTransactionsServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package org.folio.service.transactions;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import org.folio.rest.client.CirculationStorageModuleClient;
+import org.folio.rest.client.FeesFinesModuleClient;
+import org.folio.rest.client.UserModuleClient;
+import org.folio.rest.jaxrs.model.User;
+import org.folio.rest.jaxrs.model.UserTransactions;
+import org.folio.rest.util.OkapiConnectionParams;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class OpenTransactionsServiceImpl implements OpenTransactionsService {
+
+  public static final String LOANS = "loans";
+  public static final String REQUESTS = "requests";
+  public static final String ACCOUNTS = "accounts";
+  public static final String MANUAL_BLOCKS = "manualblocks";
+  public static final String PROXIES = "proxies";
+  private CirculationStorageModuleClient circulationClient;
+  private FeesFinesModuleClient feesFinesClient;
+  private UserModuleClient userClient;
+
+  public OpenTransactionsServiceImpl(CirculationStorageModuleClient circulationClient, FeesFinesModuleClient feesFinesClient, UserModuleClient userClient) {
+    this.circulationClient = circulationClient;
+    this.feesFinesClient = feesFinesClient;
+    this.userClient = userClient;
+  }
+
+  @Override
+  public Future<UserTransactions> getTransactionsOfUser(User user, OkapiConnectionParams connectionParams) {
+    String id = user.getId();
+    Future<Map.Entry<String, Integer>> loansFuture =
+      circulationClient.lookupOpenLoansByUserId(id, connectionParams)
+        .map(loans -> loans.orElse(Collections.emptyList()))
+        .map(loans -> Map.entry(LOANS, loans.size()));
```

Comment:
Why does it fetch loan records if all we need is the count? The loan records are fetched and then thrown away.
Consider using the `limit=0` feature of RMB that returns the exact (!) count without fetching any record:
https://github.com/folio-org/raml-module-builder#estimated-totalrecords

Reply:
I switched to use RMB's `limit=0` feature for fetching the exact counts of open transactions.
I wasn't aware of that. Thanks for pointing me to that!